### PR TITLE
Client Side: Business Hours block from George's Random Blocks

### PR DESCRIPTION
**Pull request by roccotripaldi**
#### Changes proposed in this Pull Request

* Adds client-side code for the business hours block from https://github.com/georgestephanis/random-blocks

#### Testing instructions

Testing instructions:

1. use this [jurasic.ninja link,](https://jurassic.ninja/create/?gutenpack&shortlived&jetpack-beta&calypsobranch=try/jetpack-business-hours-block&branch=try/business-hours-block) which spins up a site with this wp-calypso branch running along side this [jetpack branch](https://github.com/Automattic/jetpack/pull/10998) 
2. In the post editor on your site, try publishing a post with the Business Hours block

Editor:
screen shot 2019-02-04 at 5 42 55 pm

Front End:

screen shot 2019-02-01 at 3 29 17 pm

Fixes the post editor, which may seem broken to business owners who have no way to add business hours to their web site.

### CODE REVIEW

File: client/gutenberg/extensions/business-hours/components/HoursList.jsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+/**
+ * Internal dependencies
+ */
+
+import HoursRow from './HoursRow';
+import apiFetch from '@wordpress/api-fetch/build/index';
+
+const defaultData = {
+	days: {
+		Sun: 'Sunday',
```
Comment (enejb) [inline client/gutenberg/extensions/business-hours/components/HoursList.jsx]: The Days will probably need to be Translated.

File: client/gutenberg/extensions/business-hours/components/HoursRow.jsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+		const { days } = data;
+		return (
+			<Fragment>
+				<dt className={ day }>{ days[ day ] }</dt>
+				{ edit || ( hours[ day ].opening && hours[ day ].closing ) ? (
+					<dd className={ day }>
+						{ edit ? (
+							<TextControl
+								type="time"
+								label={ __( 'Opening', 'random-blocks' ) }
```
Comment (enejb) [inline client/gutenberg/extensions/business-hours/components/HoursRow.jsx]: `'random-blocks'` 🐟

File: client/gutenberg/extensions/business-hours/components/HoursRow.jsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+								} }
+							/>
+						) : (
+							hours[ day ].opening
+						) }
+						&nbsp;&mdash;&nbsp;
+						{ edit ? (
+							<TextControl
+								type="time"
+								label={ __( 'Closing', 'random-blocks' ) }
```
Comment (enejb) [inline client/gutenberg/extensions/business-hours/components/HoursRow.jsx]: 🐟

File: client/gutenberg/extensions/business-hours/components/HoursRow.jsx [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+										},
+									} );
+								} }
+							/>
+						) : (
+							hours[ day ].closing
+						) }
+					</dd>
+				) : (
+					<dd className={ day + ' closed' }>{ __( 'CLOSED', 'random-blocks' ) }</dd>
```
Comment (enejb) [inline client/gutenberg/extensions/business-hours/components/HoursRow.jsx]: 🐟

File: client/gutenberg/extensions/business-hours/index.js [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+import HoursList from './components/HoursList';
+
+/**
+ * Block Registrations:
+ */
+
+export const name = 'business-hours';
+
+export const settings = {
+	title: __( 'Business Hours', 'random-blocks' ),
```
Comment (enejb) [inline client/gutenberg/extensions/business-hours/index.js]: 🐟

File: client/gutenberg/extensions/business-hours/index.js [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+	icon: (
+		<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24">
```
Comment (enejb) [inline client/gutenberg/extensions/business-hours/index.js]: Lets use the `import renderMaterialIcon from 'gutenberg/extensions/presets/jetpack/utils/render-material-icon'; `here

File: client/gutenberg/extensions/business-hours/index.js [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+			<path fill="none" d="M0 0h24v24H0V0z" />
+			<path d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm.5-13H11v6l5.25 3.15.75-1.23-4.5-2.67z" />
```
Comment (enejb) [inline client/gutenberg/extensions/business-hours/index.js]: And use the <Path> gutenberg component.

File: client/gutenberg/extensions/presets/jetpack/index.json [hunk 8]
```diff
@@ -10,6 +10,7 @@
     "related-posts",
     "subscriptions",
     "tiled-gallery",
-    "vr"
+    "vr",
+	"business-hours"
```
Comment (enejb) [inline client/gutenberg/extensions/presets/jetpack/index.json]: indentation.

**Comment by enejb** [comment]
Just as and FYI 
screen shot 2018-12-19 at 12 12 46 am
This is the google UI for adding business hours. 
cc: @thomasguillot

**Comment by thomasguillot** [comment]
> Just as and FYI
> This is the google UI for adding business hours.
> cc: @thomasguillot

And its a great UI, thank you for sharing @enejb.

I think for added flexibility we could use something very similar here, see: https://cloudup.com/cWS02r0gu1k

**Comment by roccotripaldi** [comment]
@thomasguillot - in the interest of keeping things moving a long, we are going to address the design feedback in an other PR, after merging this one.

I've tested with Enej's recent changes, and it is working well

**Pull request merged**